src/index.js

```diff
@@ -120,12 +125,17 @@ module.exports = async (options) => {
         https: {
             rejectUnauthorized: !ignoreSslErrors,
         },
+        http2: useHttp2,
     };
 
     if (json && !decodeBody) {
         throw new Error('If the "json" parameter is true, "decodeBody" must be also true.');
     }
 
+    if (useCaseSensitiveHeaders && useHttp2) {
+        throw new Error('Headers must be lowercase when using http2. Turn off "useCaseSensitiveHeaders" or "useHttp2"');
```

Comment:
Technically, all lower case headers are also "case sensitive". Maybe we should automatically lowercase headers when HTTP2 is used?

Reply:
It happens automatically in got. we only just can't call the case-sensitive hook. Actually we could do it the same way as in requestAsBrowser. Just disable it without any warning 😄

Reply:
Yeah, technically, for HTTP1 you must have the headers capitalized and for HTTP2 you must have them lowercased, so having the option to mix that up doesn't really make sense 😄